main.go

```diff
@@ -192,6 +211,57 @@ func listSecrets(waitGroup *sync.WaitGroup, kubeClientset *kubernetes.Clientset)
 	}
 }
 
+func watchNamespaces(waitGroup *sync.WaitGroup, kubeClientset *kubernetes.Clientset, factory informers.SharedInformerFactory, stopper chan struct{}) {
+	log.Info().Msg("Watching for new namespaces...")
+
+	namespacesInformer := factory.Core().V1().Namespaces().Informer()
+	namespacesInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			namespace, ok := obj.(*v1.Namespace)
+			if !ok {
+				log.Warn().Msg("Watcher for namespaces returns event object of incorrect type")
+				return
+			}
+			// compare CreationTimestamp and controllerStartTime and act only on latest events
+			isNewNamespace := namespace.CreationTimestamp.Sub(controllerStartTime).Seconds() > 0
+			if isNewNamespace {
+
+				log.Info().Msg("Listing secrets with 'copyToAllNamespaces' for all namespaces...")
+
+				secrets, err := kubeClientset.CoreV1().Secrets("").List(metav1.ListOptions{})
```

Comment:
You can filter on label, might it be possible to do a similar thing on annotation? Otherwise we should perhaps add a label to these secrets so they can be selected more efficiently; can be done in a separate PR though.

Reply:
I did some research but couldn't find a way of filter on annotation.
So probably we will have to add it as label.